HLTrigger/HLTfilters/plugins/L1TEnergySumFilterT.h

```diff
@@ -101,23 +101,51 @@ bool L1TEnergySumFilterT<T>::hltFilter(edm::Event& iEvent,
 
   auto const& l1tSums = iEvent.getHandle(l1tSumToken_);
 
+  /// Thiago: this is now a bit ugly, because the fact that the object is HT or MHT
+  /// is now encoded on the fact that it is the 0th or 1st element in the vector.
+  ///
+  /*
+  l1t::EtSum HT = l1L1PFPhase1L1JetSums->at(0);
+  l1t::EtSum MHT = l1L1PFPhase1L1JetSums->at(1);
+  l1extra_.phase1PuppiHT = HT.pt();
+  l1extra_.phase1PuppiMHTEt = MHT.pt();
+  l1t::EtSum met = l1MetPF->at(0);
+  l1extra_.puppiMETEt = met.et();
+  */
+
   int nSum(0);
-  for (auto iSum = l1tSums->begin(); iSum != l1tSums->end(); ++iSum) {
-    double offlinePt = 0.0;
-    // pt or sumEt?
-    if (l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFMET or
-        l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFMHT) {
-      offlinePt = offlineEnergySum(iSum->pt());
-    } else if (l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFETT or
-               l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFHT) {
-      offlinePt = offlineEnergySum(iSum->sumEt());
-    }
-
-    if (offlinePt >= minPt_) {
-      ++nSum;
-      edm::Ref<std::vector<T>> ref(l1tSums, std::distance(l1tSums->begin(), iSum));
-      filterproduct.addObject(l1tSumType_, ref);
-    }
+  double onlinePt(0.0);
+  double offlinePt(0.0);
+  auto iSum = l1tSums->begin();
```

Comment:
Accessing `iSum` can lead to a crash if `*l1tSums` is empty. Since there is also a case below using `++iSum`, there should be a check that the size of `l1tSums` (the exact check depends on what makes more sense for this plugin; from the comments, it sounds like l1tSums should have size==2).

Reply:
Hm, the rabbit hole goes a bit deeper than I thought. So, l1tSums will have size==1 if it is a MET, and size==2 if it is a HT/MHT. I will add an overall check for for size>0, and a check for size>1 if we are in the HT/MHT case.